### DISCUSSION TITLE
pekwm-0.2.0

### DISCUSSION THREAD

**Discussion by pekdon** [pekwm/pekwm#113] [General]
First release in the new 0.2.X series of pekwm releases coming with quite a bit of changes and additions.

autotools are no longer supported for building pekwm, CMake is now the standard way of building pekwm but if your build environment does not have pekwm installed a simple shell configure script and a set of plain makefiles are included as well.

C++11 is no longer a requirement, compilation with GCC-2.95.3 has been tested to allow for compilation on older systems.

pekwm no longer is a window manager only distribution, it now ships with the following utilities:

* pekwm_bg, background setting utility with support for pekwm textures and possible to set from themes.
* pekwm_ctrl, wmctrl inspired utility for sending commands to pekwm from the command line.
* pekwm_dialog, xmessage inspired utility using pekwm themes. Currently used for restart dialog in case of a pekwm crash and by the pekwm_theme utility.
* pekwm_panel, simple user configurable panel application.
* pekwm_screenshot, super simple screenshot utility.
* pekwm_theme, theme management utility downloading themes from https://www.pekwm.se/themes/

See NEWS.md for more details on what has changed and look in doc/ for the most up to date documentation.

<hr /><em>This discussion was created from the release <a href='https://github.com/pekdon/pekwm/releases/tag/release-0.2.0'>pekwm-0.2.0</a>.</em>

**Comment by pclouds** [comment]
Request for packaging 0.2.0 in Gentoo Linux opened: https://bugs.gentoo.org/821184

**Reply by pekdon** [reply]
Hope that one will get some attention at some point! A handful of distributions have gotten the new release in already. :)

**Comment by biopsin** [comment]
A issue when try to crosscompile target musl, Im not shure how to takle it for the moment..
```
-- Performing Test PEKWM_HAVE_GCC_DIAGNOSTICS_PUSH
CMake Error: TRY_RUN() invoked in cross-compiling mode, please set the following cache variables appropriately:
   PEKWM_HAVE_GCC_DIAGNOSTICS_PUSH_EXITCODE (advanced)
   PEKWM_HAVE_GCC_DIAGNOSTICS_PUSH_EXITCODE__TRYRUN_OUTPUT (advanced)
For details see /builddir/pekwm-release-0.2.0/build/TryRunResults.cmake
-- Performing Test PEKWM_HAVE_GCC_DIAGNOSTICS_PUSH - Failed
-- Configuring incomplete, errors occurred!

```

**Reply by pekdon** [reply]
Interesting, need to figure out how to do that check without TRY_RUN then I suppose.

**Reply by pclouds** [reply]
Maybe this helps https://cmake.org/cmake/help/latest/command/try_run.html#behavior-when-cross-compiling

**Reply by pekdon** [reply]
You should however also have a bunch of warnings related to std::stoi, std::stof etc, don't you?

**Reply by pekdon** [reply]
@biopsin could you try cross compiling using the try_compile branch?

**Reply by biopsin** [reply]
Ok, will look into it closer this weekend, thank you so far!

**Reply by pekdon** [reply]
You could retry this on current master, at least the try run part has been removed.

**Reply by biopsin** [reply]
Heya, sorry for MIA, I'll see if I get some spare time trying again against new release soon.
Anyways thank you for your time and new release.
